### PR TITLE
Recaptcha error

### PR DESCRIPTION
**Pull request by flash1452**
Pull Request for Issue #8877 
#### Summary of Changes

Additional check introduced to check if actually an array is being processed inside foreach statement.
#### Testing Instructions

if the recaptcha is invalid during a refresh i get this error:

Warning: Invalid argument supplied for foreach()

### CODE REVIEW

**Comment by flash1452** [comment]
@brianteeman Please rview my PR

File: plugins/captcha/recaptcha/recaptcha.php [hunk 1]
```diff
@@ -197,15 +197,19 @@ private function getResponse($privatekey, $remoteip, $response, $challenge = nul
 				$reCaptcha = new JReCaptcha($privatekey);
 				$response  = $reCaptcha->verifyResponse($remoteip, $response);
 
-				if ( !isset($response->success) || !$response->success)
+				if (is_array($response->errorCodes) && is_object($response))
```
Comment (piotr-cz) [inline plugins/captcha/recaptcha/recaptcha.php]: `$response` is a _JReCaptchaResponse_ object, so no need for `is_object` check

**Comment by roland-d** [comment]
@flash1452 Can you provide some clearer test instructions? Also have a look at the comment by @piotr-cz Thanks.

File: plugins/captcha/recaptcha/recaptcha.php (hunk 1 above)
Comment (flash1452) [inline plugins/captcha/recaptcha/recaptcha.php]: @piotr-cz I did some research in this matter and found a link in stackoverflow - [this](http://stackoverflow.com/questions/2630013/invalid-argument-supplied-for-foreach). I think its the cleanest way of verifying.

Comment (piotr-cz) [inline plugins/captcha/recaptcha/recaptcha.php]: I think you should check for the `success` property first and then test if it's possible to itinerate over `errorCodes`

**Comment by brianteeman** [comment]
@flash1452 please can you respond or this will have to be closed in a few weeks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9305">issues.joomla.org/joomla-cms/9305</a>.</sub>

**Comment by brianteeman** [comment]
@flash1452 I am closing this at this time due to lack of response to multiple requests for test instructions. It can always be re-opened if they are provided.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9305">issues.joomla.org/joomla-cms/9305</a>.</sub>

**Pull request closed without merging**